### PR TITLE
fix @problame's diagnostics complaints + store Position in AsciiFile

### PR DESCRIPTION
**Pull request by problame**
See https://github.com/comprakt/comprakt/pull/18#issuecomment-435030893 for my review of the original diagnostics PR.
I'll do a perf regression test on storing the ref in Position and post it here...

### CODE REVIEW

**Comment by reiner-dolp** [comment]
I don't see why one variant should be better than the other.

**Comment by reiner-dolp** [comment]
You have to fix the clippy warnings first.

**Comment by problame** [comment]
```
 sudo perf stat --repeat 4 ./comprakt_pre --lextest  /tmp/testfile > /dev/null

 Performance counter stats for './comprakt_pre --lextest /tmp/testfile' (4 runs):

       1586,536498      task-clock (msec)         #    1,000 CPUs utilized            ( +-  0,22% )
                 5      context-switches          #    0,003 K/sec                    ( +- 20,00% )
                 0      cpu-migrations            #    0,000 K/sec                    ( +-100,00% )
               236      page-faults               #    0,148 K/sec                    ( +-  0,37% )
     4.114.459.428      cycles                    #    2,593 GHz                      ( +-  0,22% )
     7.913.689.079      instructions              #    1,92  insn per cycle           ( +-  0,03% )
     1.479.797.972      branches                  #  932,722 M/sec                    ( +-  0,03% )
        13.766.541      branch-misses             #    0,93% of all branches          ( +-  0,28% )

           1,58691 +- 0,00350 seconds time elapsed  ( +-  0,22% )



sudo perf stat --repeat 4 ./comprakt_post --lextest  /tmp/testfile > /dev/null

 Performance counter stats for './comprakt_post --lextest /tmp/testfile' (4 runs):

       1588,008826      task-clock (msec)         #    1,000 CPUs utilized            ( +-  0,17% )
                11      context-switches          #    0,007 K/sec                    ( +- 66,36% )
                 0      cpu-migrations            #    0,000 K/sec                  
               236      page-faults               #    0,149 K/sec                    ( +-  0,17% )
     4.116.260.831      cycles                    #    2,592 GHz                      ( +-  0,16% )
     7.905.290.815      instructions              #    1,92  insn per cycle           ( +-  0,02% )
     1.477.867.784      branches                  #  930,642 M/sec                    ( +-  0,03% )
        13.761.536      branch-misses             #    0,93% of all branches          ( +-  0,59% )

           1,58854 +- 0,00282 seconds time elapsed  ( +-  0,18% )
```

**Comment by problame** [comment]
OK, we don't have any significant change in cache misses or run time, probably because the release binary doesn't move the tokens around. Or because the prefetchers are really good.
@reiner-dolp ok?

File: src/main.rs [hunk 1]
```diff
@@ -4,11 +4,13 @@
 #![feature(bind_by_move_pattern_guards)]
 #![feature(const_str_as_bytes)]
 #![feature(box_syntax)]
+#![feature(repeat_generic_slice)]
```
Comment (reiner-dolp) [inline src/main.rs]: where do we need this? Wouldn't a `iter::repeat(X).take(N).collect()` (maybe with flatten) work?

Comment (problame) [reply src/main.rs]: in the asciifile tests. You used repeat for str, and I changed everything to byte strings `b""` => needed repeat thre

**Comment by reiner-dolp** [comment]
bors try

**Pull request merged**